R/score.R

```diff
@@ -106,7 +115,11 @@ score <- function(data,
                   ...) {
 
   # preparations ---------------------------------------------------------------
-  check_data <- check_forecasts(data)
+  if (is.scoringutils_check(data)) {
```

Comment:
is this an s3 method or not? If it is then fine (but why call the non-default?) if it isn't I think it needs renaming.

Reply:
I might be deeply confused about what an S3 method is... 
But data should have a class "scoringutils_check" that we can check for. 
And if I look at something like `is.data.table` then it essentially is also just `function (x) inherits(x, "data.table")`, which is broadly similar to the function here (could use 'x' instead of 'data' though)

Reply:
So its not an s3 method for is and neither is is.data.table. 

I think we shouldn't use the is. naming scheme here as it doeesn't match package convention. What about just swapping out to is_scoringutils_check?

Reply:
`is_scoringutils_check()` sounds good to me. I'll update the PR. 

How exactly can one tell / control whether something is an S3 method? We have a similar looking function, 
`print.scoringutils_check <- function(x, ...) { }` which is an S3 method, right? 
If you call print() on it, then it calls print.scoringutils_check(). 

I recognise that here we're calling `print()` whereas in the other example we're not calling `is()`. But how does R know internally? Just by using a standard from of `defaultfunction.somenewclass(x, ...) { }`?

Reply:
Claude would like to tell you some things :) 

Here is some more detail on what makes a function a generic in S3:

- It calls `UseMethod()` - This is the key thing that makes a function a generic. `UseMethod()` does the method dispatch to find the right method to call.

- It is named after the generic it defines - By convention, a generic has the same name as the methods it dispatches to. For example, `print()` is the generic that dispatches to methods like `print.data.frame()`.

- It has minimal computation - The generic itself shouldn't contain much logic, since the real computation happens in the methods. The generic just calls `UseMethod()` then returns what the method returns.

- It defines an interface - A generic defines the set of arguments that methods for associated classes should share. This defines a common interface for interacting with different classes in a generic way.

So in summary, a generic:

- Calls `UseMethod()` to enable dispatch
- Is named after the methods it dispatches to 
- Contains minimal logic itself
- Defines a common interface that methods implement

This is in contrast to a regular function, which contains the logic itself and doesn't dispatch methods.

https://adv-r.hadley.nz/s3.html

Reply:
I don't know why is isn't an s3 method it looks like it isn't. Take a look at the code for is and the code for print

Reply:
`is_scoringutils_check()` is perfect